File: src/services/galeria.service.ts
/**
 * Servicio de galería de fotos.
 * Listar fotos y subir nuevas imágenes.
 */
import apiClient from './api-client';
import { Foto } from '../types';

export const galeriaService = {
  /**
   * Listar fotos de una farmacia (paginadas, más recientes primero).
   */
  async listar(farmaciaId: number, page: number = 1): Promise<Foto[]> {
    const response = await apiClient.get<{ data: Foto[] }>(
      `/farmacias/${farmaciaId}/galeria`,
      { params: { page, limit: 50 } }
    );
    return response.data.data;
  },

  /**
   * Subir foto asociada a una farmacia y visita.
   */
  async subir(farmaciaId: number, imageUri: string, visitaId?: number): Promise<Foto> {
    const formData = new FormData();
    const filename = imageUri.split('/').pop() || 'photo.jpg';
    const match = /\.(\w+)$/.exec(filename);
    const type = match ? `image/${match[1]}` : 'image/jpeg';

    formData.append('foto', {
      uri: imageUri,
      name: filename,
      type,
    } as unknown as Blob);

    if (visitaId) {
      formData.append('visita_id', String(visitaId));
    }

    const response = await apiClient.post<{ data: Foto }>(
      `/farmacias/${farmaciaId}/galeria`,
      formData,
      { headers: { 'Content-Type': 'multipart/form-data' } }
    );
    return response.data.data;
  },
};

File: lib/services/galeria_service.dart
import 'package:dio/dio.dart';
import '../core/api_client.dart';
import '../models/foto.dart';

/// Servicio de galería de fotos.
class GaleriaService {
  /// Listar fotos de una petición (paginadas).
  static Future<GaleriaResponse> listar(int peticionId, {int page = 1}) async {
    final response = await apiClient.get(
      '/galeria/list',
      queryParameters: {'peticion_id': peticionId, 'page': page},
    );
    return GaleriaResponse.fromJson(response.data['data'] as Map<String, dynamic>);
  }

  /// Subir foto asociada a una petición.
  static Future<int> subir(
    int peticionId,
    String filePath, {
    String tipo = 'foto_visita',
  }) async {
    final fileName = filePath.split('/').last;

    final formData = FormData.fromMap({
      'foto': await MultipartFile.fromFile(filePath, filename: fileName),
      'peticion_id': peticionId.toString(),
      'tipo': tipo,
    });

    final response = await apiClient.post(
      '/galeria/upload',
      data: formData,
      options: Options(headers: {'Content-Type': 'multipart/form-data'}),
    );

    return response.data['data']['foto_id'] as int;
  }

  /// Obtener URL de thumbnail para una foto.
  static String getThumbUrl(int fotoId) {
    return '$apiBaseUrl/galeria/thumb?id=$fotoId';
  }

  /// Obtener URL completa para una foto.
  static String getFullUrl(int fotoId) {
    return '$apiBaseUrl/galeria/thumb?id=$fotoId&full=1';
  }
}
